From 8d05c13f1de2dd807ca6e35e2161c85e9506123d Mon Sep 17 00:00:00 2001
From: Sylwester Nawrocki <s.nawrocki@samsung.com>
Date: Wed, 6 Feb 2013 18:34:42 +0100
Subject: [PATCH 0168/1302] ARM: dts: exynos4: Update i2c-isp node in
 exynos4412-slp_pq.dts

Update compatible property and add pinctrl properties for ISP I2C1
device node.

ARM: dts: Add camera device nodes for PQ board

This patch adds all nodes for camera devices on an example Exynos4412 SoC
based board. This is all what's required in the board dts file to enable
rear facing camera (S5C73M3 sensor).

The aliases node contains entries required for the camera processing
data path entity drivers.

The sensor nodes use standard port/remote-endpoint nodes convention.
Internal SoC links between entities are not specified this way and
are coded in the driver instead.

The S5C73M3 sensor uses two control buses: I2C and SPI. There are
two, i2c_0 and spi_1 bus controller child nodes assigned to it.

Signed-off-by: Sylwester Nawrocki <s.nawrocki@samsung.com>
Signed-off-by: Kyungmin Park <kyungmin.park@samsung.com>
Signed-off-by: MyungJoo Ham <myungjoo.ham@samsung.com>
---
 arch/arm/boot/dts/exynos4412-slp_pq.dts | 39 ++++++++++++++++++---------------
 1 file changed, 21 insertions(+), 18 deletions(-)

diff --git a/arch/arm/boot/dts/exynos4412-slp_pq.dts b/arch/arm/boot/dts/exynos4412-slp_pq.dts
index 23bc1ac..a832bc2 100644
--- a/arch/arm/boot/dts/exynos4412-slp_pq.dts
+++ b/arch/arm/boot/dts/exynos4412-slp_pq.dts
@@ -651,23 +651,6 @@
 		};
 	};
 
-	i2c0_isp: i2c-isp@12130000 {
-		s5k6a3@10 {
-			compatible = "samsung,s5k6a3";
-			reg = <0x10>;
-			svdda-supply = <&cam_io_reg>;
-			svddio-supply = <&ldo19_reg>;
-			clock-frequency = <24000000>;
-			gpios = <&gpm1 6 0>;
-			port {
-				is_s5k6a3_ep: endpoint {
-					remote-endpoint = <&csis1_ep>;
-					data-lanes = <1>;
-				};
-			};
-		};
-	};
-
 	camera {
 		status = "okay";
 
@@ -735,7 +718,7 @@
 
 		fimc-is@12000000 {
 			status = "okay";
-			pinctrl-0 = <&fimc_is_i2c0 &fimc_is_i2c1 &fimc_is_uart>;
+			pinctrl-0 = <&fimc_is_uart>;
 			pinctrl-names = "default";
 			memory-region = <&fimc_mem>;
 
@@ -746,6 +729,26 @@
 			fimc_lite_1: fimc-lite@123A0000 {
 				status = "okay";
 			};
+
+			i2c1_isp: i2c-isp@12140000 {
+				pinctrl-0 = <&fimc_is_i2c1>;
+				pinctrl-names = "default";
+
+				s5k6a3@10 {
+					compatible = "samsung,s5k6a3";
+					reg = <0x10>;
+					svdda-supply = <&cam_io_reg>;
+					svddio-supply = <&ldo19_reg>;
+					clock-frequency = <24000000>;
+					gpios = <&gpm1 6 0>;
+					port {
+						is_s5k6a3_ep: endpoint {
+							remote-endpoint = <&csis1_ep>;
+							data-lanes = <1>;
+						};
+					};
+				};
+			};
 		};
 	};
 
-- 
1.8.3.2

